rest/blip_client_test.go

```diff
@@ -571,74 +573,194 @@ func getCollectionsForBLIP(_ testing.TB, rt *RestTester) []string {
 	return collections
 }
 
-func createBlipTesterClientOpts(tb testing.TB, rt *RestTester, opts *BlipTesterClientOpts) (client *BlipTesterClient, err error) {
+func NewBlipTesterClientOpts(opts *BlipTesterClientOpts) (client *BlipTesterClient) {
 	if opts == nil {
 		opts = &BlipTesterClientOpts{}
 	}
-	btc := BlipTesterClient{
+	client = &BlipTesterClient{
 		BlipTesterClientOpts: *opts,
-		rt:                   rt,
 	}
 
+	return client
+}
+
+func (btc *BlipTesterClient) Close() {
+	btc.pullReplication.Close()
+	btc.pushReplication.Close()
+	for _, collectionClient := range btc.collectionClients {
+		collectionClient.Close()
+	}
+	if btc.nonCollectionAwareClient != nil {
+		btc.nonCollectionAwareClient.Close()
+	}
+}
+
+func (btc *BlipTesterClient) RunDualClients(btc2 *BlipTesterClient, test func(t *testing.T), t *testing.T, rtConfig *RestTesterConfig) {
```

Comment:
I think we might be able to take this refactoring a little bit further to simplify testing.

Right now, there's nothing forcing a dev writing a test to use the Run methods other than convention (which is easily forgotten).

It would be handy to refactor this more to force us to put test code inside the function called by Run.
- I'm thinking of something like moving the methods like `PutRev`, `StartOneshotPull`, `WaitForVersion`, etc. onto a type that is only accessible from the function passed into Run.
  - (This type might actually be called `BlipTesterClient`, and the "test runner" called something else)
  - Right now it's easy to forget to use `Run()`, since autocomplete will suggest you can do things like `btc.PutRev` outside of the context of a `btc.Run(...)`
- By doing this, it might allow you to do something clever like creating your clients in the test function itself, rather than having to write something like `RunDualClients` (what happens when we test 3 clients with HLVs? :) )